service/storage_service.cc

```diff
@@ -1121,20 +1125,27 @@ class topology_coordinator {
                 auto node = get_node_to_work_on(std::move(guard));
 
                 // make sure all nodes know about new topology (we require all nodes to be alive for topo change for now)
-                std::tie(node, exec_command_res) = co_await exec_global_command(
-                        std::move(node), raft_topology_cmd{raft_topology_cmd::command::barrier}, false);
-                if (!exec_command_res) {
-                    break;
+                {
+                    auto f = co_await coroutine::as_future(exec_global_command(
+                        std::move(node), raft_topology_cmd{raft_topology_cmd::command::barrier}, false));
+                    if (f.failed()) {
+                        slogger.error("raft topology: transition_state::write_both_read_old, "
+                                      "raft_topology_cmd::command::barrier failed, error {}",
+                                      f.get_exception());
+                        break;
+                    }
```

Comment:
Missing `node = std::move(f).get()`?
But I wonder how is it possible that the tests passed with this missing...

Reply:
> But I wonder how is it possible that the tests passed with this missing...

`node.rs->state == node_state::removing` part of the `if` which is next to this code just executed `raft_topology_cmd::command::stream_ranges` without guard, the `else` part does `release_node` at the beginning of `exec_direct_command`.

I'll fix this in the follow-up.